hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/keygen/TestCustomKeyGenerator.java

```diff
@@ -317,7 +333,7 @@ public void testPartitionFieldsInImproperFormat(TypedProperties props) {
       keyGenerator.getPartitionPath(row);
       Assertions.fail("should fail when partition key field is provided in improper format!");
     } catch (Exception e) {
-      Assertions.assertTrue(getNestedConstructorErrorCause(e).getMessage().contains("Unable to find field names for partition path in proper format"));
+      assertTrue(getNestedConstructorErrorCause(e).getMessage().contains("Unable to find field names for partition path in proper format"));
```

Comment:
Instead of asserting exception message, can we reuse existing Hoodie exception or create a new one and assert that exception is thrown? I know this is not related to your change but would be better to do that in a separate PR.